src/getter.rs

```diff
@@ -524,4 +524,42 @@ impl Getter for JavaCode {
             _ => SpaceKind::Unknown,
         }
     }
+
+    fn get_op_type(node: &Node) -> HalsteadType {
+        use Java::*;
+        // Some guides that informed grammar choice for Halstead
+        // keywords, operators, literals: https://docs.oracle.com/javase/specs/jls/se18/html/jls-3.html#jls-3.12
+        // https://www.geeksforgeeks.org/software-engineering-halsteads-software-metrics/?msclkid=5e181114abef11ecbb03527e95a34828
+        let typ = node.object().kind_id();
+
+        match typ.into() {
+            // Operator: function calls
+            MethodInvocation
+            // Operator: control flow
+            | If | Else | Switch | Case | Try | Catch | Throw | Throws | Throws2 | For | While | Continue | Break | Do | Finally
+            // Operator: keywords
+            | New | Return | Default | Abstract | Assert | Instanceof | Extends | Final | Implements | Transient | Synchronized | Super | This | VoidType
+            // Operator: brackets and comma and terminators (separators)
+            | SEMI | COMMA | COLONCOLON | LBRACE | LBRACK | LPAREN | RBRACE | RBRACK | RPAREN | DOTDOTDOT | DOT
+            // Operator: operators
+            | EQ | LT | GT | BANG | TILDE | QMARK | COLON // no grammar for lambda operator ->
+            | EQEQ | LTEQ | GTEQ | BANGEQ | AMPAMP | PIPEPIPE | PLUSPLUS | DASHDASH
+            | PLUS | DASH | STAR | SLASH | AMP | PIPE | CARET | PERCENT| LTLT | GTGT | GTGTGT
+            | PLUSEQ | DASHEQ | STAREQ | SLASHEQ | AMPEQ | PIPEEQ | CARETEQ | PERCENTEQ | LTLTEQ | GTGTEQ | GTGTGTEQ
+            // type identifier
+            | TypeIdentifier | IntegralType | FloatingPointType | BooleanType
+            => {
+                HalsteadType::Operator
+            },
+            // Operands: variables, constants, literals
+            Identifier | NullLiteral | ClassLiteral | StringLiteral | CharacterLiteral | HexIntegerLiteral | OctalIntegerLiteral | BinaryIntegerLiteral | DecimalIntegerLiteral | HexFloatingPointLiteral | DecimalFloatingPointLiteral  => {
```

Comment:
Can we split this long line one more lines?

Reply:
I did find it weird the formatter didn't do that. Done 👍